### PR TITLE
feat: added support for multiple cron jobs in a class

### PR DESCRIPTION
**Pull request by dragondungeon1**
Since my last update i am converting all my annotations to attributes.

Then I found a file that has 3 cron jobs, so I tried to add 3 attribute cron jobs but since I forgot to add IS_REPEATABLE it doesn't work. 

So now i am adding it.

I will also put in a PR for the master branch.

### CODE REVIEW

**Comment by Padam87** [comment]
https://github.com/Padam87/CronBundle/blob/master/Tests/HelperTest.php#L64

Works just fine in the test suite.

I see, you mean in one declaration. I never use attributes that way, but ok :)

Please add a test for it too.

File: Annotation/Job.php [hunk 1]
```diff
@@ -9,7 +9,8 @@
  * @NamedArgumentConstructor
  * @Target("CLASS")
  */
-#[\Attribute(\Attribute::TARGET_CLASS)]
+#[\Attribute(\Attribute::TARGET_CLASS,)]
+#[\Attribute(\Attribute::IS_REPEATABLE,)]
```
Comment (mwansinck) [inline Annotation/Job.php]: Please remove the trailing comma's

**Review by mwansinck — CHANGES_REQUESTED** [review]

File: Annotation/Job.php (hunk 1 above)
Comment (dragondungeon1) [reply Annotation/Job.php]: check

**Review by mwansinck — APPROVED** [review]

**Comment by dragondungeon1** [comment]
@Padam87 
 In the PR before this you told me that ->newInstance() also could work.
 
newInstance is checking if an attribute is repeatable or not and will fail if it isnt, resulting in your current test for TwoAttributeCommand failing.

This wasn't possible earlier because of the way we handled the Annotations, but with the NamedArgumentConstructor Annotation and the followed modifications we now can use newInstance()


This is how i will be using the attributes: 
image

**Comment by Padam87** [comment]
So far it passes without a problem, adding 2 jobs to the tab.

Please provide a failing test warranting the new instance change, and a new test covering the repeatable feature.

**Comment by dragondungeon1** [comment]
>  Please provide a failing test warranting the new instance change

I don't understand what you are trying to achieve here, the newInstance() method shall not fail in any way I can predict. (it will fail by removing Attribute::IS_REPEATABLE from the Job class, but this can't be achieved at runtime, because this is "coded")

>  and a new test covering the repeatable feature.

I believe the TwoAttributesCommand test is already covering this. When removing the Attribute::IS_REPEATABLE this test will then fail (because the newInstance() will fail to execute

**Comment by Padam87** [comment]
> I don't understand what you are trying to achieve here, the newInstance() method shall not fail in any way I can predict. (it will fail by removing Attribute::IS_REPEATABLE from the Job class, but this can't be achieved at runtime, because this is "coded")

You are trying to replace ->newInstance with this: https://github.com/Padam87/CronBundle/pull/17/commits/9e52580a6dc29ca73cc54fc74c1a1823e3407ba1
I don't understand why, and I asked for a failing test warranting this change.

>I believe the TwoAttributesCommand test is already covering this. When removing the Attribute::IS_REPEATABLE this test will then fail (because the newInstance() will fail to execute

The test passes without IS_REPEATABLE.
https://github.com/Padam87/CronBundle/runs/7594044255?check_suite_focus=true#step:5:12

**Comment by Padam87** [comment]
Ah, man, you didn't replace with newInstance the last time,  and this time you didn't rebase your code on top of the new one... that is what the confusion is.

Next time please don't mark it as resolved until it actually is, because I don't have the time to review every small thing.

Closing this, and doing it proper.

**Comment by dragondungeon1** [comment]
Dear Adam, thank you for bringing my changes to the codebase.

Next time will be better

**Pull request closed without merging**